crates/rpc/rpc-builder/src/lib.rs

```diff
@@ -1166,29 +1191,56 @@ impl RpcServerConfig {
         self
     }
 
-    /// Configure the cors domains for http _and_ ws
-    pub fn with_cors(self, cors_domain: Option<String>) -> Self {
-        self.with_http_cors(cors_domain.clone()).with_ws_cors(cors_domain)
+    /// Configures the ws server
+    ///
+    /// Note: this always configures an [`EthSubscriptionIdProvider`] [`IdProvider`] for
+    /// convenience. To set a custom [`IdProvider`], please use [`Self::with_id_provider`].
+    pub fn with_ws(mut self, config: ServerBuilder<Identity, Identity>) -> Self {
```

Comment:
these accept self and return Self, and don't care about the current generic, so these should be moved to the 

`impl<RpcMiddleware> RpcServerConfig<RpcMiddleware>

scope